website/pages/restaurants/[id]/index.tsx

```diff
@@ -1,118 +1,424 @@
+import supabase from "@/utils/supabase";
 
-import React, { useEffect, useRef, useState } from "react";
+import Image from "next/image";
+import React, { useEffect, useState } from "react";
 
 import { Autoplay, EffectFade } from "swiper/modules";
 import { Swiper, SwiperSlide } from "swiper/react";
 
-import supabase from "@/utils/supabase";
-import SectionTitle from "@/components/Common/SectionTitle";
-
 import "swiper/css";
 import "swiper/css/effect-fade";
+import { InView } from "react-intersection-observer";
 import { useRouter } from "next/router";
 import RootLayout from "@/pages/layout";
+import NotFound from "@/components/NotFound";
 
-const Restaurant = () => {
+const RestaurantMenu = () => {
   const router = useRouter();
   const { id } = router.query;
+  const suffix = id?.toString().substring(id?.lastIndexOf("-") + 1);
 
-  const [isRestaurantsLoading, setIsRestaurantsLoading] = useState(true);
-  const [restaurants, setRestaurantsData] = useState<any[]>([]);
+  const [isRestaurantLoading, setIsRestaurantLoading] = useState(true);
+  const [restaurantData, setRestaurantData] = useState<any | null>(null);
+  const [menuData, setMenuData] = useState<any[]>([]);
 
   useEffect(() => {
-    const fetchData = async () => {
-      const { data, error } = await supabase
-        .from("restaurants")
-        .select()
-        .contains("tags", [(id as string)?.replace(/-/g, " ")]);
+    const fetchRestaurantData = async () => {
+      if (suffix) {
+        try {
+          const { data, error } = await supabase
+            .from("restaurants")
+            .select("*")
+            .eq("id", atob(suffix!))
+            .single();
+
+          if (error) throw error;
+
+          setRestaurantData(data);
+        } catch (error) {
+          console.error("Error fetching restaurant data:", error);
+        } finally {
+          setIsRestaurantLoading(false);
+        }
+      }
+    };
+
+    const fetchDishes = async () => {
+      if (suffix) {
+        try {
+          const { data: menusData, error } = await supabase
+            .from("menus")
+            .select("id, name")
+            .eq("restaurant_id", atob(suffix!));
 
-      if (error) throw error;
+          if (error) return error;
 
-      setRestaurantsData(data);
-      setIsRestaurantsLoading(false);
+          const dishes = await Promise.all(
+            menusData.map(async (menu) => {
+              const { data: dishData, error: dishError } = await supabase
+                .from("dishes")
+                .select("id, name, description, price, images, video")
+                .eq("menu_id", menu.id);
+
+              if (dishError) {
+                throw dishError;
+              }
+
+              return {
+                ...menu,
+                dishes: dishData,
+              };
+            })
+          );
+
+          setMenuData(dishes);
+        } catch (error) {
+          console.error("Error fetching dishes data:", error);
```

Comment:
I think the error should redirect to something went wrong page.

Reply:
- i think it would be scray moment when user gets something went wrong page. Instead we have managed error to show empty view or no view when gets error for a better experience.
- isn't good idea?

Reply:
yeah, it would be better.